snuba/query/processors/granularity_processor.py

```diff
@@ -39,3 +41,53 @@ def process_query(self, query: Query, query_settings: QuerySettings) -> None:
                 Literal(None, granularity),
             )
         )
+
+
+PERFORMANCE_GRANULARITIES = [(60, 1), (3600, 2), (86400, 3)]
+DEFAULT_MAPPED_GRANULARITY_ENUM = 1
+
+
+class MappedGranularityProcessor(QueryProcessor):
+    """
+    Use the granularity set on the query to filter on the granularity column,
+    supporting generic-metrics style enum mapping
+    """
+
+    def __init__(
+        self,
+        accepted_granularities: Sequence[Tuple[int, int]],
+        default_granularity_enum: int,
+    ):
+        self._accepted_granularities = accepted_granularities
+        self._available_granularities_values = [
+            x[0] for x in self._accepted_granularities
+        ]
+        self._default_granularity_enum = default_granularity_enum
+
+    def __get_granularity(self, query: Query) -> int:
+        """Find the best fitting granularity for this query"""
+        requested_granularity = query.get_granularity()
+
+        if requested_granularity is None:
+            return self._default_granularity_enum
+        elif requested_granularity > 0:
+            for (granularity, mapped_value) in reversed(self._accepted_granularities):
+                if (
+                    requested_granularity % granularity == 0
+                    or requested_granularity == mapped_value
```

Comment:
Why is this second condition needed?

Reply:
it's probably not, but we do want product to eventually own handling this mapping so I thought it could be a good way to transition